bookings/views.py

```diff
@@ -1,3 +1,84 @@
-from django.shortcuts import render
+import json
+from datetime         import datetime
 
-# Create your views here.
+from django.views     import View
+from django.http      import JsonResponse
+from django.db.models import Q
+
+from .models          import Booking
+from products.models  import Product
+from users.utils      import login_decorator
+
+class BookingView(View):
+    @login_decorator
+    def post(self, request, product_id):
+        try:
+            data        = json.loads(request.body)
+            
+            check_in    = data.get("checkin")
+            check_out   = data.get("checkout")
+            head_count  = data.get("count")
+            
+            if not check_in or not check_out or not head_count:
+                raise KeyError
```

Comment:
`query paramter` 중에 하나라도 없으면 keyError를 raise하는 경우라고 하면 data에서 접근해서 가져올때부터 `[]`로 접근하면 코드를 줄여볼 수 있을 것 같네요 

```suggestion
            check_in    = data["checkin"]
            check_out   = data["checkout"]
            head_count  = data["count"]
            
            if not check_in or not check_out or not head_count:
                raise KeyError
```